product/crud/handlers/handle_delete_product.py

```diff
@@ -1,47 +1,38 @@
-from http import HTTPStatus
-from typing import Any, Dict
+from http import HTTPMethod, HTTPStatus
 
 from aws_lambda_env_modeler import get_environment_variables, init_environment_variables
+from aws_lambda_powertools.logging import correlation_paths
 from aws_lambda_powertools.metrics import MetricUnit
-from aws_lambda_powertools.utilities.parser import ValidationError, parse
 from aws_lambda_powertools.utilities.typing import LambdaContext
 
 from product.crud.domain_logic.delete_product import delete_product
+from product.crud.handlers.constants import PRODUCT_PATH
 from product.crud.handlers.schemas.env_vars import DeleteVars
-from product.crud.handlers.utils.http_responses import build_response
 from product.crud.handlers.utils.observability import logger, metrics, tracer
-from product.crud.schemas.exceptions import InternalServerException
+from product.crud.handlers.utils.rest_api_resolver import app
 from product.crud.schemas.input import DeleteProductRequest
 
 
-@init_environment_variables(model=DeleteVars)
-@metrics.log_metrics
-@tracer.capture_lambda_handler(capture_response=False)
-def handle_delete_product(event: Dict[str, Any], context: LambdaContext) -> Dict[str, Any]:
-    logger.set_correlation_id(context.aws_request_id)
+@app.route(PRODUCT_PATH, method=HTTPMethod.DELETE)
```

Comment:
```suggestion
@app.delete(PRODUCT_PATH)
```

there's a shortcut ;)